nat-lab/guidelines.md

```diff
@@ -0,0 +1,215 @@
+### Natlab Guidelines:
+
+1. **Use Parametrization:**
+   - Test across all operating systems when possible.
+   - Ensure parametrization for each node.
+
+   **Why:**
+   - Covers more areas.
+   - Better readability.
+   - Wider usage of a test.
+
+   **Example:**
+   ```
+   @pytest.mark.parametrize("alpha_node", [ConnectionTag.MAC_VM, ConnectionTag.WINDOWS_VM, ConnectionTag.DOCKER_CONE_CLIENT_1])
+   @pytest.mark.parametrize("beta_node", ConnectionTag.MAC_VM)
+   def test_params(alpha_node, beta_node):
+   ```
+
+2. **Assertions Clarity:**
+   - Assertions must be explicit and easily visible in the test.
+   - Avoid assertions in different threads, keep it within test, raise an exception instead if needed.
```

Comment:
Correct me if I'm wrong but I think that 'thread' is not correct here - I think 'task' is the python name for what you mean here.

Reply:
true that, will change